packages/loader/container-loader/src/container.ts

```diff
@@ -859,6 +859,21 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                             cancel: this.closeSignal,
                         }, // progress
                     );
+
+                    switch (runResult.status) {
+                        case "succeeded":
+                            this.service = runResult.result;
+                            break;
+                        case "failed":
+                            throw runResult.error;
+                        case "aborted":
+                            throw new GenericError(
+                                // We know the reason since we use closeSignal above but don't pass an onRetry callback
+                                `Container.attach() was aborted because the container was closed`,
```

Comment:
I'd suggest returning this error only if container is indeed closed, and return more generic messages otherwise, to ensure that we know (and have time to address it) if second case ever happened.
This function can go further and not care that much about how call failed if container is closed. So, it may blindly return this error even if we get 403 / 404 or something else.